ENGAGEHF/Bluetooth/MeasurementManager.swift

```diff
@@ -0,0 +1,230 @@
+//
+// This source file is part of the ENGAGE-HF project based on the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import BluetoothServices
+import FirebaseAuth
+import FirebaseCore
+import FirebaseFirestore
+import Foundation
+import HealthKit
+import OSLog
+import Spezi
+@_spi(TestingSupport) import SpeziBluetooth
+import SpeziFirestore
+
+
+// Functionality:
+// - Store the user's measurement history in an array
+//      - Watch for changes in the user's blood pressure and weight measurement collections in firebase
+// - Convert an incoming measurement into a HKSample and transform to FHIR Observation
+// - Save a given measurement to Firebase
+@Observable
+class MeasurementManager: Module, EnvironmentAccessible {
+    private static var _manager: MeasurementManager?
+    static var manager: MeasurementManager {
+        guard let manager = _manager else {
+            fatalError("Accessing shared MeasurmentManager before initialized.")
+        }
+        return manager
+    }
+    
+    @ObservationIgnored @StandardActor private var standard: ENGAGEHFStandard
+    private let logger = Logger(subsystem: "ENGAGEHF", category: "MeasurementManager")
+    
+    var deviceInformation: DeviceInformationService?
+    var weightScaleParams: WeightScaleFeature?
+    var deviceName: String?
+    
+    var newMeasurement: HKQuantitySample?
+    
+    var showSheet = false
+    
+    
+    init() {
+        MeasurementManager._manager = self
+    }
+    
+    
+    // Called to reset measurement manager after taking a measurement
+    func clear() {
+        self.showSheet = false
+        self.newMeasurement = nil
+        self.deviceInformation = nil
+        self.weightScaleParams = nil
+        self.deviceName = nil
+    }
+    
+    
+    // Called by WeightScaleDevice on change of WeightMeasurement Characteristic
+    func loadMeasurement(_ measurement: WeightMeasurement) {
+        // Convert to HKQuantitySample after downloading from Firestore
+        let convertedMeasurement = convertToHKSample(measurement)
+        
+        // Save the sample to the Measurement Manager
+        self.newMeasurement = convertedMeasurement
```

Comment:
Could be compressed in a single line:
```suggestion
        self.newMeasurement = convertToHKSample(measurement)
```